### DISCUSSION TITLE
onBlur Question

### DISCUSSION THREAD

**Discussion by battebery** [nick-keller/react-datasheet-grid#64] [Q&A] [answered]
Hello,

I have a use case - trying to figure out how to write updates on each edit.  I'm maintaining the table data in state, and was hoping to pull out the updated row on edit based on the updated cell data.  In my data each row is a record in the database, so updating by row is useful.

Trying to capture the last edited cell with onBlur but it seems inconsistent in the cell data it gets.  I was expecting after leaving the cell that onBlur would receive the last cell you were editing. Is this a bug in onBlur or am I confusing how it should work?

> **If I [enter] to leave a cell after editing I get the correct last active cell data in onBlur** 
> i.e. if I'm editing {row :0, col: 0} and [enter] I'm now in {row :1, col: 0} and the cell data passed to onBlur is {row :0, col: 0} so I can know what cell was updated.
> 
> **If I [tab] or [arrowkey] out after edit I get the newly active cell data, not the last active** 
> i.e. if I'm editing {row :0, col: 0} and [tab] i'm now in {row :0, col: 1} but the cell data passed to onBlur is {row :0, col: 1} so I'm not able to know what cell was updated.  Same for arrow keys.

Is there a better way for me to identify which cell was last updated or a better way overall that I'm missing to implement my use case?

Thanks.

**Comment by dmikoss** [comment]
I think we need some way to determine **incremental** updates on table cells (edits, pasting) with  full access to row data.

**Reply by battebery** [reply]
Good point, I agree it would be useful to somehow capture/return a set of all row/cells updated post-edit to support other editing modes.  
My initial use case attempts were admittedly narrow focused on single cell edits and trying to understand if onBlur was working correctly.  I didn't really see anything else to hook into to know when the user was done editing to trigger saves.

**Comment by nick-keller** [comment]
@battebery sorry for the delay, what you are describing sounds like a bug. I'll test it out and create an issue if I can reproduce.

Now regarding your broader problem, the way I would approach it is to keep in memory the original data and compare it to the updated data to see which rows have been updated. This sounds like an easier / less prone to bugs solution.

@dmikoss can you be more specific, what would you need exactly? This is definitely an important topic, I'm trying to understand what people need and what would be the best way to achieve it.

Thanks to both of you for your feedbacks ;)

**Reply by nick-keller** [reply]
This has been solved in 3.3.2

**Comment by nick-keller** [comment, marked as answer]
This is now available in 3.4.0, please read the doc for [`onChange`](https://react-datasheet-grid.netlify.app/docs/api-reference/props#onchange)
